test/testcil.pl

```diff
@@ -657,7 +658,7 @@ sub addToGroup {
 addBadComment("scott/globalprob", "Notbug. Not a bug if fails on a non-Linux machine ;-)");
 addTest("scott/bisonerror $gcc");
 addTest("scott/cmpzero");
-addTest("scott/kernel1 $gcc");
+# addTest("scott/kernel1 $gcc");
```

Comment:
Why is this test no longer of interest? I would guess McPeak had a reason for adding it? In what way does it fail if it is left there?

Reply:
It fails on GCC 14 with some of the new error types that aren't so easy to suppress in backwards-compatible way.

The main line of the test is
```c
DECLARE_WAIT_QUEUE_HEAD(log_wait);
```
which comes from the Linux kernel, where `DECLARE_WAIT_QUEUE_HEAD` is supposed to be a macro (from some unknown kernel version).
This test doesn't have the macro or any includes, so this is being parsed as some very old C function declaration that declares a function without specifying its return type and also declares a new type `log_wait` in the middle of the function declaration.

So this test has never been about parsing actual kernel sources (which have the macro that expands to something else entirely). In it, CIL is just successfully parsing some _unpreprocessed_ code in a completely different way from its original meaning.
CIL will still support these old C function declarations, it's just that this test isn't enabled any more.

This could be made conditional to not run on GCC 14+ (or run with different `CFLAGS`), but I think detecting the GCC version from `gcc --version` in `Makefile` is awfully error-prone, so it's not worth the trouble for what is a misguided test to begin with.